src/main/java/com/spotify/folsom/retry/RetryingClient.java

```diff
@@ -35,23 +36,39 @@
  */
 public class RetryingClient implements RawMemcacheClient {
 
+  private final int maxRetries;
   private final RawMemcacheClient delegate;
 
   public RetryingClient(final RawMemcacheClient delegate) {
+    this(delegate, 1);
+  }
+
+  public RetryingClient(final RawMemcacheClient delegate, final int maxRetries) {
     this.delegate = delegate;
+    this.maxRetries = maxRetries;
   }
 
   @Override
-  public <T> ListenableFuture<T> send(final Request<T> request) {
-    final ListenableFuture<T> future = delegate.send(request);
-    return Futures.catchingAsync(future, MemcacheClosedException.class,
-        new AsyncFunction<MemcacheClosedException, T>() {
-          @Override
-          public ListenableFuture<T> apply(final MemcacheClosedException e) {
-            if (delegate.isConnected()) {
-              return delegate.send(request);
+  public <T> CompletableFuture<T> send(final Request<T> request) {
```

Comment:
This seems like a significant refactoring - why? Can we do it as a separate change or is it required for the java8 migration?